node_modules/oae-version/lib/api.js

```diff
@@ -0,0 +1,188 @@
+ * Each generated build has a `build-info.json` file that contains information about the build.
+ * Parse it and return the relevant version information
+ *
+ * @param  {String}     directory           The directory where the build info can be found
+ * @param  {Function}   callback            Standard callback function
+ * @param  {Object}     callback.err        An error that occurred, if any
+ * @param  {String}     callback.version    The version information for the directory
+ * @api private
+ */
+function getBuildVersion(directory, callback) {
+    var buildInfoPath = path.join(directory, 'build-info.json');
+    fs.readFile(buildInfoPath, function(err, buildInfo) {
+        if (err) {
+            return callback({'code': 500, 'msg': 'Unable to read the build info file'});
+        }
+
+        try {
+            buildInfo = JSON.parse(buildInfo);
+        } catch (ex) {
+            log().error({
+                'err': ex,
+                'path': buildInfoPath
+            }, 'Unable to parse the build info file');
+            return callback({'code': 500, 'msg': 'Unable to parse the build info file'});
+        }
+
+        return callback(null, buildInfo.version);
+    });
+}
+
+/**
+ * Get the version from git
+ *
+ * @param  {String}     directory           The directory to the git repository
+ * @param  {Function}   callback            Standard callback function
+ * @param  {Object}     callback.err        An error that occurred, if any
+ * @param  {String}     callback.version    The version information for the directory
+ * @api private
+ */
+function getGitVersion(directory, callback) {
+    gift.init(directory, function(err, repo) {
+        if (err) {
+            log().error({
+                'err': err,
+                'path': directory
+            }, 'Could not open the git repo to get the version information');
+            return callback({'code': 500, 'msg': 'Could not open the git repo'});
+        }
+
+        // Get the last tag info
+         repo.tags(function(err, tags) {
+            // Sort them descending (last tag first)
+            tags = tags.sort(function(a, b) {
+                return b.commit.committed_date.getTime() - a.commit.committed_date.getTime();
+            });
+
+            // Get the most recent tag
+            var lastTag = tags[0];
+
+            // Get the current branch info
+            repo.branch(function(err, branch) {
+                if (err) {
+                    log().error({
+                        'err': err,
+                        'path': directory
+                    }, 'Could not get the current git branch for the version information');
+                    return callback({'code': 500, 'msg': 'Could not get the branch information'});
+                }
+
+                var msg = 'Last tag: %s, Current branch: %s, Commit SHA1: %s, Date: %s';
+                var version = util.format(msg, lastTag.name, branch.name, branch.commit.id, branch.commit.committed_date);
```

Comment:
We generate the version in the tarballs from git information, and we have a syntax for non-tag version, which includes the commit hash, the base tag, and how many commits we are beyond that. I think we should use it instead.